python/ray/autoscaler/local/onprem_server.py

```diff
@@ -0,0 +1,229 @@
+import logging
+import threading
+from http.server import SimpleHTTPRequestHandler, HTTPServer
+import yaml
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def runner_handler(node_scheduler):
+    class Handler(SimpleHTTPRequestHandler):
+        """A custom handler for LocalNodeProviderServer.
+
+        Handles all requests and responses coming into and from the
+        remote "local" node provider.
+        """
+
+        def _do_header(self, response_code=200, headers=None):
+            """Sends the header portion of the HTTP response.
+            Args:
+                response_code (int): Standard HTTP response code
+                headers (list[tuples]): Standard HTTP response headers
+            """
+            if headers is None:
+                headers = [("Content-type", "application/json")]
+
+            self.send_response(response_code)
+            for key, value in headers:
+                self.send_header(key, value)
+            self.end_headers()
+
+        def do_HEAD(self):
+            """HTTP HEAD handler method."""
+            self._do_header()
+
+        def do_GET(self):
+            """Processes requests from local node providers."""
+            if self.headers["content-length"]:
+                raw_data = (self.rfile.read(
+                    int(self.headers["content-length"]))).decode("utf-8")
+                logger.info("I got a response, content: " + str(raw_data))
+                request = json.loads(raw_data)
+                if request["request_type"] == "get_node_ips":
+                    response = node_scheduler.get_node_ips(
+                        request["num_workers"], request["cluster_name"])
+                elif request["request_type"] == "release_cluster":
```

Comment:
Why have calls like release_cluster instead of getting individual nodes, etc? That would be simpler and more flexible since it would support autoscaling automatically.